File: Notification System/src/PushNotifier.cpp
#include "PushNotifier.hpp"
#include <iostream>

void PushNotifier::sendNotification(const std::string& sensorId,
                                    const std::string& message,
                                    const std::string& timestamp) {
    std::cout << "[PUSH] " << timestamp
              << " | Capteur: " << sensorId
              << " | Message: " << message << std::endl;
    // TODO: implémenter notification desktop réelle
}

File: Notification System/notifiers/push_notifier.py
import subprocess
from .notifier import Notifier


class PushNotifier(Notifier):

    def send_notification(self, sensor_id: str, message: str, timestamp: str):
        print(f"[PUSH] {timestamp} | Capteur: {sensor_id} | Message: {message}")

        zenity_message = (
            f"Capteur : {sensor_id}\n"
            f"Message : {message}\n"
            f"Heure : {timestamp}"
        )

        try:
            subprocess.Popen([
                "zenity",
                "--info",
                "--title=Notification système",
                f"--text={zenity_message}",
                "--timeout=10"   # ⏱️ fermeture auto après 5 secondes
            ])
        except FileNotFoundError:
            print("❌ Zenity n'est pas installé sur le système")
